homeassistant/components/prana/coordinator.py

```diff
@@ -0,0 +1,84 @@
+"""Coordinator for Prana integration.
+
+Responsible for polling the device REST endpoints and normalizing data for entities.
+"""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from prana_local_api_client.exceptions import (
+    PranaApiCommunicationError,
+    PranaApiUpdateFailed,
+)
+from prana_local_api_client.prana_api_client import PranaLocalApiClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, PranaFanType, PranaSensorType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PranaCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Universal coordinator for Prana (fan, switch, sensor, light data)."""
+
+    def __init__(
+        self, hass: HomeAssistant, entry: ConfigEntry, prana_config: str | None
+    ) -> None:
+        """Initialize the Prana data update coordinator."""
+        super().__init__(
+            hass,
+            logger=_LOGGER,
+            name=f"{DOMAIN} coordinator",
+            update_interval=timedelta(seconds=10),
+            update_method=self._async_update_data,
+            config_entry=entry,
+        )
+        self.entry = entry
+        self.max_speed: int | None = None
+
+        host = self.entry.data.get("host")
+        if not host:
+            raise ValueError("Host is not specified in the config entry data")
```

Comment:
```suggestion
        host = self.entry.data[CONF_HOST]
```
we know it has to be there